### PR TITLE
Changed the way we simulate empty interfaces in juniper

### PR DESCRIPTION
**Pull request by stephanerobert**
Empty interfaces in juniper are not shown through the interface.
Fake-switch now replicate this behaviour.

### CODE REVIEW

File: tests/juniper_qfx_copper/juniper_qfx_copper_protocol_test.py [hunk 1]
```diff
@@ -363,7 +364,7 @@ def test_trunk_mode_does_not_allow_no_vlan_members(self):
 </error-message>
   <error-severity>error</error-severity>
   <error-path>
-[edit interfaces ge-0/0/3 unit 0 family]
+[self.edit interfaces ge-0/0/3 unit 0 family]
```
Comment (stephanerobert) [inline tests/juniper_qfx_copper/juniper_qfx_copper_protocol_test.py]: ho... that is wierd

File: tests/juniper/juniper_base_protocol_test.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 from tests.util.global_reactor import juniper_switch_ip, juniper_switch_netconf_port
 
 
-class JuniperBaseProtocolTest(unittest.TestCase):
+class JuniperBaseProtocolTest(BaseJuniper):
     def setUp(self):
         self.nc = self.create_client()
```
Comment (fbouliane) [inline tests/juniper/juniper_base_protocol_test.py]: This should be moved to BaseJuniper,
It uses it and depends on the children creating this variable.

File: tests/juniper_qfx_copper/juniper_qfx_copper_protocol_test.py [hunk 3]
```diff
@@ -1148,10 +1149,9 @@ def test_assign_port_to_aggregated_interface(self):
         assert_that(ge001.xpath("ether-options/*"), has_length(1))
         assert_that(ge001.xpath("ether-options/speed/ethernet-10g"), has_length(1))
 
-        ge002 = self.get_interface("ge-0/0/2")
-        assert_that(ge002.xpath("*"), has_length(1))
-        assert_that(ge002.xpath("unit"), has_length(0))
-        assert_that(ge002.xpath("ether-options"), has_length(0))
+        with self.assertRaises(Exception) as exc:
+            self.get_interface("ge-0/0/2", )
```
Comment (fbouliane) [inline tests/juniper_qfx_copper/juniper_qfx_copper_protocol_test.py]: Real code does not raise an exception. I think this should check the state of the xml.

File: tests/juniper/juniper_base_protocol_test.py [hunk 4]
```diff
@@ -1168,10 +1168,9 @@ def test_assign_port_to_aggregated_interface(self):
         assert_that(ge001.xpath("ether-options/*"), has_length(1))
         assert_that(ge001.xpath("ether-options/speed/ethernet-10g"), has_length(1))
 
-        ge002 = self.get_interface("ge-0/0/2")
-        assert_that(ge002.xpath("*"), has_length(1))
-        assert_that(ge002.xpath("unit"), has_length(0))
-        assert_that(ge002.xpath("ether-options"), has_length(0))
+        with self.assertRaises(Exception) as exc:
```
Comment (fbouliane) [inline tests/juniper/juniper_base_protocol_test.py]: same comment here https://github.com/internap/fake-switches/pull/57/files#r67959231

File: tests/juniper/__init__.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    return m
+
+
+def reset_interface(interface_name):
+    def m(edit):
+        edit({"interfaces": {
+            "interface": [{XML_ATTRIBUTES: {"operation": "replace"}},
+                          {"name": interface_name},
+                          {"native-vlan-id": ""},
```
Comment (fbouliane) [inline tests/juniper/__init__.py]: Not supported on juniper Ex2200.

**Comment by fbouliane** [comment]
This is the diff between the refactored function for Juniper vs qfx:
https://gist.github.com/fbouliane/f72f97eba5303e63e52dbe3b1ac5e65b
As may be seen we need a way to remove the native_vlan_id for juniper and letting them there for qfx.

**Comment by stephanerobert** [comment]
Reclosing this pendora box... Please recheck

**Comment by godp1301** [comment]
👍

**Comment by JoProvost** [comment]
Please fix the commit message for Fakw-switch :)

**Comment by fbouliane** [comment]
:+1:

**Pull request merged**